json2cmake/__init__.py

```diff
@@ -69,7 +72,9 @@ def read(self, input):
         for entry in database:
             command = freeze(
                 parsecommand(entry['command'], directory=entry['directory']))
-            self.targets.setdefault(command, set()).add(entry['file'])
+            file = os.path.relpath(
+                os.path.join(entry['directory'], entry['file']), os.getcwd())
```

Comment:
I can see how the previous code is broken for targets with the same filename in different directories, but I think this code is broken for absolute filenames.